make/common/JdkNativeCompilation.gmk

```diff
@@ -76,18 +76,158 @@ endif
 GetJavaHeaderDir = \
   $(if $(strip $1),$(wildcard $(SUPPORT_OUTPUTDIR)/headers/$(strip $1)))
 
-# Process a dir description such as "java.base:headers" into a set of proper absolute paths.
-ProcessDir = \
+HOTSPOT_INCLUDE_DIR := $(TOPDIR)/src/hotspot/share/include \
+    $(TOPDIR)/src/hotspot/os/$(HOTSPOT_TARGET_OS_TYPE)/include
+
+# Preprocess a source dir destination. Pass on absolute paths unchanged. Expand
+# description such as "java.base:headers" into a set of proper absolute paths,
+# by looking in $MODULE/[shared|$OS|$OSTYPE]/native.
+# Treat *:libjvm as a special case, and replace it with the proper hotspot
+# include directories.
+ProcessSrcDir = \
   $(if $(findstring :, $1), \
-    $(call FindSrcDirsForComponent, $(firstword $(subst :, , $1)), $(lastword $(subst :, , $1))) \
+    $(if $(filter %:libjvm, $1), \
+      $(HOTSPOT_INCLUDE_DIR) \
+    , \
+      $(call FindSrcDirsForComponent, $(firstword $(subst :, , $1)), \
+          $(lastword $(subst :, , $1))) \
+    ) \
   , \
     $(if $(filter /%, $1), \
       $1 \
     , \
-      $(call FindSrcDirsForComponent, $(MODULE), $1) \
+      $(if $(filter libjvm, $1), \
+        $(HOTSPOT_INCLUDE_DIR) \
+      , \
+        $(call FindSrcDirsForComponent, $(MODULE), $1) \
+      ) \
     ) \
   )
 
+# Create a proper LIBPATH for the given library.
+# $1: The name of the rule (namespace)
+# $2: The safe namespace of the library
+define ResolveLibPath
```

Comment:
This macro is a horrible mess. But, sort of the point of this PR was to collect all this horrible mess in a single place, instead of having this strewn out all over the code base. 

My intention is to make the placement of libraries during build more systematic in a future PR, so this function can be simplified.